plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityAttributeModifiers.java

```diff
@@ -175,13 +217,19 @@ public String getPropertyId() {
     // These can be modified by such mechanisms as <@link mechanism EntityTag.attribute_base_values>, <@link mechanism EntityTag.attribute_modifiers>, <@link mechanism EntityTag.add_attribute_modifiers>,
     // <@link mechanism EntityTag.remove_attribute_modifiers>, <@link mechanism ItemTag.attribute_modifiers>, <@link mechanism ItemTag.add_attribute_modifiers>, <@link mechanism ItemTag.remove_attribute_modifiers>, ...
     //
-    // The input format of each of the 'add' and set mechanisms is slightly complicated:  a MapTag where the keys are attribute names, and values are a ListTag of modifiers,
-    // where each modifier is itself a MapTag with required keys 'operation' and 'amount', and optional keys 'name', 'slot', and 'id'.
+    // The input format of each of the 'add' and set mechanisms is slightly complicated: a MapTag where the keys are attribute names, and values are a ListTag of modifiers,
+    // where each modifier is itself a MapTag with required keys 'operation' and 'amount', and additionally:
+    // Before MC 1.20.6: optional 'name', 'slot', and 'id' keys.
+    // On MC 1.20.6: optional 'name', 'slot_group', and 'id' keys.
+    // The default ID will be randomly generated, the default name will be the attribute name.
+    // After MC 1.21: required 'key' key, and optional 'slot_group'.
+    // The 'key' is the attribute's name/identifier in a "namespaced:key" format (defaulting to the "minecraft" namespace), which has to be unique from other modifiers on the object.
     //
     // Valid operations: ADD_NUMBER, ADD_SCALAR, and MULTIPLY_SCALAR_1
-    // Valid slots: HAND, OFF_HAND, FEET, LEGS, CHEST, HEAD, ANY
+    // Valid slots (used up to MC 1.20.6): HAND, OFF_HAND, FEET, LEGS, CHEST, HEAD, ANY
+    // Valid slot groups (used on MC 1.20.6+): <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/inventory/EquipmentSlotGroup.html>

```

Comment:
SlotGroup looks like just an extension to slot. We don't need to match this underlying change into scripts - just keep it as `slot` and allow the new options

Reply:
Combined into just a `slot` key with legacy input support for `EquipmentSlot` on modern versions